### DISCUSSION TITLE
Support Deno

### DISCUSSION THREAD

**Discussion by idranme** [vuejs/rfcs#429] [General]
Deno has higher performance.

[https://deno.land/](https://deno.land/)

**Comment by darrylnoakes** [comment]
Support Deno in which way? Higher performance than what?

Deno is a backend runtime, like Node.js. Vue.js is a frontend framework, for use in browsers.
Node.js/Deno are fundamentally different to browsers. They only way the Vue.js would be used with them really (aside using them for the server, which is completely different and unrelated) would be for SSR/SSG/etc.

Are you meaning support it for SSR and such?

**Comment by jods4** [comment]
Dupe of #278 ?

**Comment by leopiccionia** [comment]
Deno supports web-compatible [`ReadableStream`](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream), so SSR should, in theory, work.

[Vue SFC Playground](https://sfc.vuejs.org/) can compile SFC files in the browser, so it probably could compile them in Deno, too.

The hardest part would probably be importing `*.vue` files inside JS/TS files in Deno. As far as I know, neither TypeScript nor SWC expose APIs for implementing custom loaders; I'm not sure if import maps could help.

**Comment by yyx990803** [comment]
There's not much actionable here. If you mean you want Vue to be able to SSR in Deno, it already works (by bundling `@vue/server-renderer`'s ESM build, which is runtime-agnostic, into your app).

**Reply by idranme** [reply]
I hope that the example on the official website can give information about Deno to promote the ecological development of Deno.

**Reply by yyx990803** [reply]
It's not really Vue's responsibility to promote a specific JavaScript runtime.
